tests/ManageClientTest.php

```diff
@@ -53,4 +55,72 @@ function (RequestInterface $request): ResponseInterface {
         self::assertSame('111', $sandboxes[0]->getId());
         self::assertSame('222', $sandboxes[1]->getId());
     }
+
+    public function testListProjects(): void
+    {
+        $guzzleHandler = HandlerStack::create(new MockHandler([
+            function (RequestInterface $request): ResponseInterface {
+                self::assertSame('GET', $request->getMethod());
+                self::assertSame('manage-token', $request->getHeaderLine('X-KBC-ManageApiToken'));
+                self::assertSame('/manage/projects', $request->getUri()->getPath());
+                self::assertSame('hasPersistentStorage=1&nextPageToken=', $request->getUri()->getQuery());
+
+                return new Response(200, [], (string) json_encode([
+                    'data' => [
+                        [
+                            'id' => '111',
+                            'projectId' => 'project1',
+                            'tokenId' => 'token-id',
+                            'type' => 'python',
+                            'active' => true,
+                            'createdTimestamp' => '2021-12-12T12:00:00Z',
+                        ],
```

Comment:
if these have persistent storage, then I guess they should return it?  I know it's just a mock, ale ... 😄

Reply:
I don't really have any reasons here :slightly_smiling_face: On one hand, you're righ it does not represent how project looks in reality. But on the other hand, this shows the minimal needed configuration for the code to work and adding the persistent storage to data may lead you to a wrong assumption that the data actually matters.

Reply:
Ok, after thinking of it for 30 more second, I'll add it :smile:

Reply:
Done